pkg/progress/supermin.go

```diff
@@ -0,0 +1,301 @@
+ln -s /proc/self/fd/0 /dev/stdin
+ln -s /proc/self/fd/1 /dev/stdout
+ln -s /proc/self/fd/2 /dev/stderr
+
+# need /dev/shm for podman
+mkdir -p /dev/shm
+mount -t tmpfs tmpfs /dev/shm
+
+# osbuild needs /run
+mkdir -p /run
+mount -t tmpfs tmpfs /run
+
+### from here on we diverge from the above core-assembler script
+
+# ensure network
+/usr/sbin/dhclient eth0
+
+# central for osbuild
+mknod /dev/loop-control c 10 237
+
+# map in the host /output dir, virtiofsd support uid remapping so this
+# should be fine
+mkdir -p /output
+mount -t virtiofs osbuild_output /output
+
+# We cannot put the host osbuild "store" on our /store dir even with
+# virtiofsd as there will be issues with uid remapping and selinux
+# labels.
+mkdir -p /host/store
+mount -t virtiofs osbuild_store /host/store
+# rsync/cp the caches so that our build is faster
+echo "Populate /store from host"
+mkdir /store
+rsync -av --exclude=./tmp /host/store/ /store
+
+# fetch/cache sources first so that the host cache gets updated
+# XXX: fix hardcoded /output/manifest.json
+echo "Fetching sources"
+osbuild \
+  --cache /store \
+  /output/manifest.json
+# rsync 
+rsync -av --exclude=./tmp /store/ /host/store
+
+# XXX: an error will currently trigger a kernel painc (pid=0 died)
+# XXX2: fix hardcoded /output/manifest.json
+echo "Running osbuild"
+osbuild \
+  --export %s \
+  --output-directory /output \
+  --cache /store \
+  /output/manifest.json
+
+# trigger clean shutdown via sysreq
+echo _suo > /proc/sysrq-trigger
+# shutdown is async so we need to sleep here or PID=0 dies
+sleep 999
+`
+
+func addInitTar(superminDir, superminInitScript string) error {
+	initTarF, err := os.Create(filepath.Join(superminDir, "init.tgz"))
+	if err != nil {
+		return err
+	}
+	defer initTarF.Close()
+	initTar := tar.NewWriter(initTarF)
+	defer initTar.Close()
+	if err := initTar.WriteHeader(&tar.Header{
+		Name: "init",
+		Size: int64(len(superminInitScript)),
+		Mode: 0755,
+	}); err != nil {
+		return err
+	}
+	if _, err := initTar.Write([]byte(superminInitScript)); err != nil {
+		return err
+	}
+	return nil
+}
+
+func superminPrepare(prepareDir, export string) error {
+	superminInitScript := fmt.Sprintf(superminInitScriptFmt, export)
+
+	err := util.RunCmdSync(
+		"supermin", "--prepare", "--use-installed",
+		// fundamental
+		"util-linux",
+		// convenient
+		"rsync",
+		// basic networking
+		"ca-certificates", "dhcp-client", "iproute",
+		// loop-device support
+		"kernel-modules",
+		// osbuild and friends
+		"osbuild", "osbuild-depsolve-dnf", "osbuild-lvm2", "osbuild-luks2", "osbuild-ostree",
+		// lvm
+		"lvm2",
+		// target dir
+		"-o", prepareDir,
+	)
+	if err != nil {
+		return fmt.Errorf("supermin prepare failed: %w", err)
+	}
+	if err := addInitTar(prepareDir, superminInitScript); err != nil {
+		return err
+	}
+	return nil
+}
+
+func superminBuild(prepareDir, buildDir string) error {
+	err := util.RunCmdSync(
+		"supermin",
+		"--build", prepareDir,
+		// XXX: what is the right size?
```

Comment:
As it's thinly provisioned (right?) it doesn't matter too much to have it be large.

However, the real fix is to use virtiofs for the rootfs I think, which would obviate much of the role of supermin.